src/Hazelcast.Net.Tests/Messaging/ClientMessageTests.cs

```diff
@@ -152,15 +152,20 @@ public void Enumerator()
                 Assert.That(!e.NextIsNotTheEnd());
 
                 e.MoveNext();
+                Assert.That(e.Current, Is.Not.Null);
                 while (!e.Current.IsBeginStruct) e.MoveNext();
-                // at 2
-                Assert.That(e.NextIsNotTheEnd());
+                // at 2, is BeginStruct
+                Assert.That(!e.AtStructEnd()); // whether e.Current is EndStruct
+                Assert.That(e.NextIsNotTheEnd()); // whether e.Current is !EndStruct
```

Comment:
I think comment should be like that;

```suggestion
                Assert.That(e.NextIsNotTheEnd()); // whether e.Current.Next is !EndStruct
```
Because `e.NextIsNotTheEnd()` is just peaking the next one, right?

Reply:
Yes. But what's tricky is that it's looking that the next one that `e.Take()` would return, and that is the `e.Current` one. It's wacky but we cannot change that enumerator as it must mimick the Java enumerator for the generated codecs to work. This is why we have this test: to help understand the strange enumerator ;)